ui/tests/integration/allocation-row-test.js

```diff
@@ -0,0 +1,85 @@
+import { getOwner } from '@ember/application';
+import { test, moduleForComponent } from 'ember-qunit';
+import wait from 'ember-test-helpers/wait';
+import hbs from 'htmlbars-inline-precompile';
+import generateResources from '../../mirage/data/generate-resources';
+import { startMirage } from 'nomad-ui/initializers/ember-cli-mirage';
+import Response from 'ember-cli-mirage/response';
+
+moduleForComponent('allocation-row', 'Integration | Component | allocation row', {
+  integration: true,
+  beforeEach() {
+    this.store = getOwner(this).lookup('service:store');
+    this.server = startMirage();
+    this.server.create('namespace');
+    this.server.create('node');
+    this.server.create('job', { createAllocations: false });
+  },
+  afterEach() {
+    this.server.shutdown();
+  },
+});
+
+test('Allocation row polls for stats, even when it errors or has an invalid response', function(assert) {
+  const component = this;
+
+  let currentFrame = 0;
+  let frames = [
+    JSON.stringify({ ResourceUsage: generateResources() }),
+    JSON.stringify({ ResourceUsage: generateResources() }),
+    null,
+    '<Not>Valid JSON</Not>',
+    JSON.stringify({ ResourceUsage: generateResources() }),
+  ];
+  const backoffSequence = [50];
+
+  this.server.get('/client/allocation/:id/stats', function() {
+    const response = frames[++currentFrame];
+
+    // Disable polling to stop the EC task in the component
+    if (currentFrame >= frames.length) {
+      component.set('enablePolling', false);
```

Comment:
I haven't seen this idea before, pretty neat. Did you consider using `cancelTimers` and prefer this instead?